src/addons/sourcemod/scripting/zr/weapons/zmarket.inc

```diff
@@ -111,19 +111,16 @@ void ZMarketLoad()
 
             if (FindCharInString(zmarketcommand, ',') != -1)
             {
-                int idx;
-                int lastidx;
-                while ((idx = FindCharInString(zmarketcommand[lastidx], ',')) != -1)
+                char commands[8][32];
+                int count = ExplodeString(zmarketcommand, ",", commands, sizeof(commands), sizeof(commands[]));
+
+                for (int i = 0; i < count; i++)
                 {
-                    char out[16];
-                    char fmt[8];
-                    Format(fmt, sizeof(fmt), "%%.%ds", idx);
-                    Format(out, sizeof(out), fmt, zmarketcommand[lastidx]);
-                    RegConsoleCmd(out, ZMarketBuyCommand, weaponname);
-                    lastidx += ++idx;
-
-                    if (FindCharInString(zmarketcommand[lastidx], ',') == -1 && zmarketcommand[lastidx+1] != '\0')
-                        RegConsoleCmd(zmarketcommand[lastidx], ZMarketBuyCommand, weaponname);
+                    TrimString(commands[i]);
+                    if (strlen(commands[i]) > 0)
+                    {
+                        RegConsoleCmd(commands[i], ZMarketBuyCommand, weaponname);
+                    }
                 }
             }
             else
```

Comment:
instead of duplicating code, you could do check before the loop if count <= 0 and put command in commands[0] so that way we have only one loop with important functions